src/Data/Validation/Semigroup.purs

```diff
@@ -70,3 +71,9 @@ instance applyV :: (Semigroup err) => Apply (V err) where
 
 instance applicativeV :: (Semigroup err) => Applicative (V err) where
   pure = Valid
+
+instance semigroupV :: (Semigroup err, Semigroup a) => Semigroup (V err a) where
+  append x y = append <$> x <*> y
```

Comment:
I guess not, but difference should be really small right?
Also javascript runtime could optimise lift2 if it's called too often.